### ISSUE TITLE
Incorrect typing of container props

### ISSUE THREAD

**Issue by jbelloin** [mbrevda/react-image#256]
Hi,

In react-image.d.ts I see the typing of container, loaderContainer and unloaderContainer is `() => JSX.Element`

As it expects a children to be passed, the correct typing should be:

```
      container?: (children: React.ReactNode) => JSX.Element
      loaderContainer?: (children: React.ReactNode) => JSX.Element
      unloaderContainer?: (children: React.ReactNode) => JSX.Element
```

On top of it, to extend all the props of the img html element you can replace ` [key: string]: any` by `export interface ImgProps extends React.ComponentPropsWithRef<'img'>`

Putting it together, it gives the following typing:

```
    export interface ImgProps extends React.ComponentPropsWithoutRef<'img'> {
      src?: string | string[]
      loader?: JSX.Element
      unloader?: JSX.Element
      decode?: boolean
      crossorigin?: string
      container?: (children: React.ReactNode) => JSX.Element
      loaderContainer?: (children: React.ReactNode) => JSX.Element
      unloaderContainer?: (children: React.ReactNode) => JSX.Element
    }
```

**Comment by mbrevda** [comment]
cc'ing @schester44 and @leopaini who worked on the original types in #243. What do you guys think?

**Comment by schester44** [comment]
yeah that looks way more sound to me.

@jbelloin Is there anything else missing that you can tell?

**Comment by jbelloin** [comment]
As far as I've seen it's the only thing missing

**Comment by mbrevda** [comment]
Thanks all for taking a look. If one of you could add what's missing via a PR we can get this published!

**Comment by jbelloin** [comment]
Hi,

I tried pushing my update to your repo on a new branch but I'm affraid I don't have the rights...

**Comment by mbrevda** [comment]
A PR is probably simplest. You can also do a minor version bump as well as
a changelog entry.

**Comment by jbelloin** [comment]
PR opened :)

**Comment by mbrevda** [comment]
fix in #258
